mmdet/ops/merge_cells.py

```diff
@@ -0,0 +1,119 @@
+from abc import abstractmethod
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from mmcv.cnn import ConvModule
+
+
+class BaseMergeCell(nn.Module):
+    """The basic class for cells used in NAS-FPN and NAS-FCOS.
+
+    BaseMergeCell takes 2 inputs. After applying concolution
+    on them, they are resized to the target size. Then,
+    they go through binary_op, which depends on the type of cell.
+
+    """
+
+    def __init__(self,
+                 in_channels=256,
+                 out_channels=256,
+                 with_conv=True,
+                 conv_cfg=dict(),
+                 norm_cfg=None,
+                 order=('act', 'conv', 'norm'),
+                 with_input1_conv=False,
+                 with_input2_conv=False,
+                 input_conv_cfg=None,
+                 input_norm_cfg=None,
+                 upsample_mode='nearest'):
+        super(BaseMergeCell, self).__init__()
+        assert upsample_mode in ['nearest', 'bilinear']
+        self.with_conv = with_conv
+        self.with_input1_conv = with_input1_conv
+        self.with_input2_conv = with_input2_conv
+        self.upsample_mode = upsample_mode
+
+        if self.with_conv:
+            self.conv_out = ConvModule(
+                in_channels,
+                out_channels,
+                kernel_size=conv_cfg['kernel_size'],
+                groups=conv_cfg['groups'],
+                bias=conv_cfg['bias'],
+                padding=conv_cfg['padding'],
+                norm_cfg=norm_cfg,
+                order=order)
+
+        self.conv1 = self._build_input_conv(
+            out_channels, input_conv_cfg,
+            input_norm_cfg) if with_input1_conv else nn.Sequential()
+        self.conv2 = self._build_input_conv(
+            out_channels, input_conv_cfg,
+            input_norm_cfg) if with_input2_conv else nn.Sequential()
+
+    def _build_input_conv(self, channel, conv_cfg, norm_cfg):
+        return ConvModule(
+            channel,
+            channel,
+            3,
+            padding=1,
+            conv_cfg=conv_cfg,
+            norm_cfg=norm_cfg,
+            bias=True)
+
+    @abstractmethod
+    def _binary_op(self, x1, x2):
+        raise NotImplementedError
```

Comment:
In an abstract method, `raise NotImplementedError` is unnecessary, you may just use `pass`.